### DISCUSSION TITLE
Phpspreadsheet surface chart generate corrupted xlsx file

### DISCUSSION THREAD

**Discussion by lvlukola** [PHPOffice/PhpSpreadsheet#2805] [Q&A] [answered]
**Please help with example surface chart**

May be my code wrong

    $dataSeriesLabels = [
            new DataSeriesValues(DataSeriesValues::DATASERIES_TYPE_STRING, null, null, 1, ['5-6']),
            new DataSeriesValues(DataSeriesValues::DATASERIES_TYPE_STRING, null, null, 1, ['6-7']),
            new DataSeriesValues(DataSeriesValues::DATASERIES_TYPE_STRING, null, null, 1, ['7-8']),
        ];

        $xAxisTickValues = [
            new DataSeriesValues(DataSeriesValues::DATASERIES_TYPE_NUMBER, null, null, 9, [1, 2, 3, 4, 5, 6, 7, 8, 9]),
        ];

        $dataSeriesValues = [
            new DataSeriesValues(DataSeriesValues::DATASERIES_TYPE_NUMBER, null, null, 9, [6, 6, 6, 6, 6, 6, 5.9, 6, 6]),
            new DataSeriesValues(DataSeriesValues::DATASERIES_TYPE_NUMBER, null, null, 9, [6, 6, 6, 6.5, 7, 7, 7, 7, 7]),
            new DataSeriesValues(DataSeriesValues::DATASERIES_TYPE_NUMBER, null, null, 9, [6, 6, 6, 7, 8, 8, 8, 8, 7.9]),
        ];

        $series = new DataSeries(
            DataSeries::TYPE_SURFACECHART,
            DataSeries::GROUPING_STANDARD,
            range(0, count($dataSeriesValues) - 1),
            $dataSeriesLabels,
            $xAxisTickValues,
            $dataSeriesValues,
            null, // plotDirection
            null, // smooth line
            DataSeries::STYLE_LINEMARKER  // plotStyle
        );

        $plotArea = new PlotArea(null, [$series]);
        $legend = new ChartLegend(ChartLegend::POSITION_BOTTOM, null, false);

        $title = new Title('График распредления температур в пределах кр');

        $chart = new Chart(
            'chart2',
            $title,
            $legend,
            $plotArea,
            true,
            DataSeries::EMPTY_AS_GAP,
        );

        $chart->setTopLeftPosition("A$1");
        $chart->setBottomRightPosition("P$20");

        $sheet->addChart($chart);

**Comment by oleibman** [comment, marked as answer]
The error is definitely in the PhpSpreadsheet code, not yours. I recently opened issue #2763, but have not had a chance to work on it.

**Reply by lvlukola** [reply]
Thanks
